frontend/src/metabase/querying/filters/utils/dates.unit.spec.ts

```diff
@@ -138,3 +138,186 @@ describe("getDateFilterClause", () => {
     },
   );
 });
+
+type DateFilterDisplayNameCase = {
+  value: DateFilterValue;
+  displayName: string;
+  withPrefix?: boolean;
+};
+
+describe("getDateFilterDisplayName", () => {
+  it.each<DateFilterDisplayNameCase>([
+    {
+      value: {
+        type: "specific",
+        operator: "=",
+        values: [new Date(2024, 2, 5)],
+        hasTime: false,
+      },
+      displayName: "March 5, 2024",
+    },
+    {
+      value: {
+        type: "specific",
+        operator: "=",
+        values: [new Date(2024, 2, 5, 10, 20)],
+        hasTime: true,
+      },
+      displayName: "March 5, 2024 10:20 AM",
+    },
+    {
+      value: {
+        type: "specific",
+        operator: "=",
+        values: [new Date(2024, 2, 5)],
+        hasTime: false,
+      },
+      withPrefix: true,
+      displayName: "On March 5, 2024",
+    },
+    {
+      value: {
+        type: "specific",
+        operator: ">",
+        values: [new Date(2024, 2, 5)],
+        hasTime: false,
+      },
+      displayName: "After March 5, 2024",
+    },
+    {
+      value: {
+        type: "specific",
+        operator: "<",
+        values: [new Date(2024, 2, 5)],
+        hasTime: false,
+      },
+      displayName: "Before March 5, 2024",
+    },
+    {
+      value: {
+        type: "specific",
+        operator: "between",
+        values: [new Date(2024, 0, 1), new Date(2024, 11, 31)],
+        hasTime: false,
+      },
+      displayName: "January 1, 2024 - December 31, 2024",
+    },
+    {
+      value: {
+        type: "specific",
+        operator: "between",
+        values: [new Date(2024, 0, 1, 10, 20), new Date(2024, 11, 31, 23, 15)],
+        hasTime: true,
+      },
+      displayName: "January 1, 2024 10:20 AM - December 31, 2024 11:15 PM",
+    },
+    {
+      value: { type: "relative", value: "current", unit: "day" },
+      displayName: "Today",
+    },
+    {
+      value: { type: "relative", value: "current", unit: "year" },
+      displayName: "This Year",
+    },
+    {
+      value: { type: "relative", value: -1, unit: "day" },
+      displayName: "Yesterday",
+    },
+    {
+      value: { type: "relative", value: -2, unit: "year" },
+      displayName: "Previous 2 Years",
+    },
+    {
+      value: {
+        type: "relative",
+        value: -3,
+        unit: "month",
+        offsetValue: -1,
+        offsetUnit: "year",
+      },
+      displayName: "Previous 3 Months, starting 1 year ago",
```

Comment:
Noun capitalization is off. There are more occurrences in this file.

```suggestion
      displayName: "Previous 3 months, starting 1 year ago",
```

Reply:
I reported it separately as P3: #52088